### ISSUE TITLE
CiviSEPA NOT compatible with CiviCRM-4.6.10

### ISSUE THREAD

**Issue by systopia** [Project60/org.project60.sepa#348]
If you're using the SEPA payment processor **do not upgrade to CiviCRM 4.6.10** (yet). They added some rather careless changes in 4.6.10 that break the payment processor. CiviCRM 4.6.9 is fine.

The problem is that the billing address is now mandatory, but CiviSEPA in the current version explicitly removes these fields from the form (because it used to be optional), see [CRM-17603](https://issues.civicrm.org/jira/browse/CRM-17603)

 This leads to the following problem:
```
Billing First Name is a required field.
Billing Last Name is a required field.
Street Address is a required field.
City is a required field.
State/Province is a required field.
Postal Code is a required field.
Country is a required field.
```

**Comment by systopia** [comment]
implementation of #333 will contain code to stop the pp from hiding the billing information

**Comment by systopia** [comment]
If you're currently having trouble with this, you might want to have a look at https://github.com/systopia/org.project60.sepa/tree/dev_4.6.10, which contains a untested(!) fixes for #333 and #350 which should also (temporarily) fix *this* issue. Test well before using this, though!

**Comment by magnolia61** [comment]
Hi Bjorn, I read into this issue but could not find the PR with the change in 4.6.10 that caused this. Can you tell me. I would like to help advocate for reverting the change because I think I agree with you it was not implemented properly.

**Comment by systopia** [comment]
There is not a single PR, but it I listed a couple of suspects in the above mentioned CRM-17603. I plan to look into this in the next weeks, but any help is welcome.

**Comment by magnolia61** [comment]
Update from the Jira issue (https://issues.civicrm.org/jira/browse/CRM-17603) on this from Alan:
*You can override the "requiredness" by overriding the payment object method "getMandatoryFields" which by default just looks in the is_required value of the corresponding field. Or you can use the form alter hook to remove that is_required value.*

**Comment by systopia** [comment]
In response to Alan:

> You can override the "requiredness" by overriding the payment object method "getMandatoryFields" which by default just looks in the is_required value of the corresponding field

That's only available from CiviCRM 4.7+, see #351 

> Or you can use the form alter hook to remove that is_required value.

That seems rather "hacky" to me, and I wouldn't want to put that in a general extension.

**Comment by systopia** [comment]
New approach merged to ``1.2.x``. Will merge to ``master`` when tested and released.

**Comment by magnolia61** [comment]
Hi Bjorn, Is the 1.2.x branch still the one I need to use when I want to help test on 4.6.10 and 4.7.4?

**Comment by systopia** [comment]
Yes, I'm only waiting for feedback on https://github.com/Project60/org.project60.sepa/releases/tag/1.2.2.beta1, but it's looking good.

4.7 might not work, they changed the payment processor structure *again*. Please post your findings!

**Comment by magnolia61** [comment]
When I use CiviSEPA 1.2.2beta1 on CiviCRM 4.6.14 on a donation form it works ok now. But when I use it in an event registration the Collection Day & Start Date have reappeared and the mandatory billing block still appears (despite the setting in civisepa settings). Probably easy fixes, not?

**Comment by systopia** [comment]
Maybe...event pages are not fully supported yet (no funding)

**Comment by magnolia61** [comment]
I have no funding to help test this extension, but am willing to help test event voluntarily.
That is my donation :-)

**Comment by systopia** [comment]
> I have no funding to help test this extension, but am willing to help test event voluntarily. That is my donation :-)

I know, and we're grateful :) I just wanted to explain to the other readers why this isn't moving forward...

**Comment by systopia** [comment]
Fix released with 1.2.2
